apps/server/src/modules/provisioning/strategy/oidc/service/oidc-provisioning.service.ts

```diff
@@ -0,0 +1,96 @@
+import { Injectable, UnprocessableEntityException } from '@nestjs/common';
+import { EntityId } from '@shared/domain';
+import { SchoolDO } from '@shared/domain/domainobject/school.do';
+import { UserDO } from '@shared/domain/domainobject/user.do';
+import { AccountSaveDto } from '@src/modules/account/services/dto';
+import { AccountUc } from '@src/modules/account/uc/account.uc';
+import { RoleService } from '@src/modules/role';
+import { RoleDto } from '@src/modules/role/service/dto/role.dto';
+import { SchoolService } from '@src/modules/school';
+import { UserService } from '@src/modules/user';
+import CryptoJS from 'crypto-js';
+import { ExternalSchoolDto, ExternalUserDto } from '../../../dto';
+
+@Injectable()
+export class OidcProvisioningService {
+	constructor(
+		private readonly userService: UserService,
+		private readonly schoolService: SchoolService,
+		private readonly roleService: RoleService,
+		private readonly accountUc: AccountUc
```

Comment:
A service can't inject a uc

Reply:
We already had this discussion, when implementing this function initially. I just moved the code here. The AccountUc has mandatory validation checks that should be done when saving a account e.g. does the user already have an account or is the email unique. The Account module needs a refactoring in this regard in order to have a service function that validates objects (maybe DOs) that are supposed to be saved.
This change however is a bigger refactoring that is a bit out of scope for this ticket. Also, the PR is already big.